src/macaron/malware_analyzer/checks/detect_malicious_metadata_check.py

```diff
@@ -257,8 +258,9 @@ def run_check(self, ctx: AnalyzeContext) -> CheckResultData:
         CheckResultData
             The result of the check.
         """
-        purl = ctx.component.purl
-        parsed_purl = PackageURL.from_string(purl)
+        parsed_purl = PackageURL.from_string(ctx.component.purl)
+        if parsed_purl.type != "pypi":
+            return CheckResultData(result_tables=[], result_type=CheckResultType.FAILED)
```

Comment:
Just to let you know, we want to use this check for other ecosystems in the future too. We can add this and I will improve this implementation in my next refactoring PR of this check.

Also, `CheckResultType.FAILED` means the check has found a malware. So the result type should be `CheckResultType.UNKNOWN` if the artifact is not a Python package.